### PR TITLE
[master] Update interceptor bbes related to `http:InterceptableService` addition

### PR DESCRIPTION
**Pull request by dilanSachi**
## Purpose
Related to https://github.com/ballerina-platform/ballerina-standard-library/issues/4401

## Automation tests
 - Unit tests 
   > Code coverage information
 - Integration tests
   > Details about the test cases and coverage

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes/no
 - Ran FindSecurityBugs plugin and verified report? yes/no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes/no

### CODE REVIEW

**Comment by dilanSachi** [comment]
Build wont pass until this is merged.
https://github.com/ballerina-platform/module-ballerina-http/pull/1606

File: examples/http-request-interceptor/http_request_interceptor.md [hunk 1]
```diff
@@ -1,6 +1,6 @@
...
+The `http:RequestInterceptor` is used to intercept the request and execute some custom logic. A `RequestInterceptor` is a service object with only one resource method, which is executed before dispatching the request to the actual resource in the target service. This resource method can have parameters just like a usual resource method in an `http:Service`. A `RequestInterceptor` can be created from a service class, which includes the `http:RequestInterceptor` service type. Then, this service object can be engaged at the listener level by using the `interceptors` field in the `http:ListenerConfiguration` or at service level by using the `http:InterceptableService` in the service definition. These accept an interceptor service object or an array of interceptor service objects as an interceptor pipeline, and the interceptors are executed in the order in which they are placed in the pipeline. Use `RequestInterceptors` to execute some common logic such as logging, header manipulation, state publishing, etc., for inbound requests.
```
Comment (praneesha) [inline examples/http-request-interceptor/http_request_interceptor.md:3]:
```suggestion
The `http:RequestInterceptor` is used to intercept the request and execute custom logic. A `RequestInterceptor` is a service object with only one resource method, which is executed before dispatching the request to the actual resource in the target service. This resource method can have parameters just like a usual resource method in an `http:Service`. 

A `RequestInterceptor` can be created from a service class, which includes the `http:RequestInterceptor` service type. Then, this service object can be engaged at the listener level by using the `interceptors` field in the `http:ListenerConfiguration` or at the service level by using the `http:InterceptableService` in the service definition. 

These accept an interceptor service object or an array of interceptor service objects as an interceptor pipeline, and the interceptors are executed in the order in which they are placed in the pipeline. Use `RequestInterceptors` to execute common logic such as logging, header manipulation, state publishing, etc., for inbound requests.
```
This paragraph is too big and should be broken down to multiple for improved clarity.

File: examples/http-response-interceptor/http_response_interceptor.md [hunk 2]
```diff
@@ -1,6 +1,6 @@
...
+The `http:ResponseInterceptor` is used to intercept the response and execute some custom logic. A `ResponseInterceptor` is a service object with a remote method called `interceptResponse`, which is executed before dispatching the response to the client. A `ResponseInterceptor` can be created from a service class, which includes the `http:ResponseInterceptor` service type. Then, this service object can be engaged at the listener level by using the `interceptors` field in the `http:ListenerConfiguration` or at service level by using the `http:InterceptableService` in the service definition. These accept an interceptor service object or an array of interceptor service objects as an interceptor pipeline, and the interceptors are executed in the order in which they are placed in the pipeline. Use `ResponseInterceptors` to execute some common logic such as logging, header manipulation, state publishing, etc., for all outbound responses.
```
Comment (praneesha) [inline examples/http-response-interceptor/http_response_interceptor.md:3]:
```suggestion
The `http:ResponseInterceptor` is used to intercept the response and execute custom logic. A `ResponseInterceptor` is a service object with a remote method called `interceptResponse`, which is executed before dispatching the response to the client. A `ResponseInterceptor` can be created from a service class, which includes the `http:ResponseInterceptor` service type. 

This service object can be engaged at the listener level by using the `interceptors` field in the `http:ListenerConfiguration` or at the service level by using the `http:InterceptableService` in the service definition. These accept an interceptor service object or an array of interceptor service objects as an interceptor pipeline and the interceptors are executed in the order in which they are placed in the pipeline. 

Use `ResponseInterceptors` to execute common logic such as logging, header manipulation, state publishing, etc., for all outbound responses.
```

This paragraph is too big and should be broken down to multiple for improved clarity.

File: examples/http-request-interceptor/http_request_interceptor.bal [hunk 3]
```diff
@@ -38,15 +38,15 @@ service class RequestInterceptor {
...
-    // the target service. Hence, they will be executed only for this particular service.
-    interceptors: [new RequestInterceptor()]
-}
-service / on interceptorListener {
+// Engage interceptors at the service level using `http:InterceptableService`. The base path of the
+// interceptor services is the same as the target service. Hence, they will be executed only for
+// this particular service.
+service http:InterceptableService / on interceptorListener {
 
+    // Creates the interceptor pipeline. Request interceptor services will be executed from head to tail.
```
Comment (TharmiganK) [inline examples/http-request-interceptor/http_request_interceptor.bal:46]: Can we also update with the improvement we did to support single interceptor as the interceptor pipeline?
```suggestion
    // Creates the interceptor pipeline. The function can return a single interceptor or an array of interceptors as the interceptor pipeline. If the interceptor pipeline is an array, then the request interceptor services will be executed from head to tail.
```

File: examples/http-request-interceptor/http_request_interceptor.bal [hunk 4]
```diff
@@ -38,15 +38,15 @@ service class RequestInterceptor {
...
+    public function createInterceptors() returns [RequestInterceptor] {
```
Comment (TharmiganK) [inline examples/http-request-interceptor/http_request_interceptor.bal:47]:
```suggestion
    public function createInterceptors() returns RequestInterceptor {
```

File: examples/http-request-interceptor/http_request_interceptor.bal [hunk 5]
```diff
@@ -38,15 +38,15 @@ service class RequestInterceptor {
...
+        return [new RequestInterceptor()];
```
Comment (TharmiganK) [inline examples/http-request-interceptor/http_request_interceptor.bal:48]:
```suggestion
        return new RequestInterceptor();
```

File: examples/http-request-interceptor/http_request_interceptor.md [hunk 6]
```diff
@@ -1,6 +1,10 @@
...
+The `http:RequestInterceptor` is used to intercept the request and execute custom logic. A `RequestInterceptor` is a service object with only one resource method, which is executed before dispatching the request to the actual resource in the target service. This resource method can have parameters just like a usual resource method in an `http:Service`. 
+
+A `RequestInterceptor` can be created from a service class, which includes the `http:RequestInterceptor` service type. Then, this service object can be engaged at the listener level by using the `interceptors` field in the `http:ListenerConfiguration` or at the service level by using the `http:InterceptableService` in the service definition.
```
Comment (TharmiganK) [inline examples/http-request-interceptor/http_request_interceptor.md:5]:
```suggestion
A `RequestInterceptor` can be created from a service class, which includes the `http:RequestInterceptor` service type. Then, this service object can be engaged at the listener level by using the `interceptors` field in the `http:ListenerConfiguration` or at the service level by declaring a `http:InterceptableService` object. 
```

File: examples/http-response-interceptor/http_response_interceptor.md [hunk 7]
```diff
@@ -1,6 +1,10 @@
...
+The `http:ResponseInterceptor` is used to intercept the response and execute custom logic. A `ResponseInterceptor` is a service object with a remote method called `interceptResponse`, which is executed before dispatching the response to the client. A `ResponseInterceptor` can be created from a service class, which includes the `http:ResponseInterceptor` service type. 
+
+This service object can be engaged at the listener level by using the `interceptors` field in the `http:ListenerConfiguration` or at the service level by using the `http:InterceptableService` in the service definition. These accept an interceptor service object or an array of interceptor service objects as an interceptor pipeline and the interceptors are executed in the order in which they are placed in the pipeline.
```
Comment (TharmiganK) [inline examples/http-response-interceptor/http_response_interceptor.md:5]:
```suggestion
This service object can be engaged at the listener level by using the `interceptors` field in the `http:ListenerConfiguration` or at the service level by declaring a `http:InterceptableService` object. These accept an interceptor service object or an array of interceptor service objects as an interceptor pipeline and the interceptors are executed in the order in which they are placed in the pipeline. 
```

File: examples/http-request-interceptor/http_request_interceptor.bal (hunk 3 above)
Comment (TharmiganK) [reply examples/http-request-interceptor/http_request_interceptor.bal:46]: Can you check the other BBEs as well? (If we have a single interceptor in an array, then we can remove the array)

Comment (dilanSachi) [reply examples/http-request-interceptor/http_request_interceptor.bal]: Updated. I dont think we dont need to update other BBEs as they have the listener level initalization and we anyway need to remove those with ongoing listener level interceptor change.

**Review by TharmiganK — DISMISSED** [review]
LGTM, lets make sure the tests are passing after updating the distribution

**Review by TharmiganK — APPROVED** [review]

**Review by ushirask — APPROVED** [review]

**Review by kalaiyarasiganeshalingam — APPROVED** [review]

**Pull request merged**